rfcs/0084-ZHF-on-master-and-freeze-dates.md

```diff
@@ -0,0 +1,104 @@
+---
+feature: nixos-freeze-dates
+start-date: 2021-01-17
+author: Jonathan Ringer (@jonringer)
+co-authors:
+related-issues: [NixOS release schedule](https://github.com/NixOS/rfcs/pull/80)
+---
+
+# Summary
+[summary]: #summary
+
+To further minimize the amount of risk going into a NixOS release, I propose
+to set freeze dates regarding changes to critical packages being merged to
+staging. In addition, I also propose to stabilize the release on master
+instead of the release branch, thus avoiding the need to backport every pull
+request, and take advantage of the stabilization that unstable normally
+receives.
+
+# Definitions
+[definitions]: #definition
+
+- "Freezing" a branch or package.
+  - Disallow breaking changes to branch or package. In SemVer terminology, disallow major version bumps.
+  - A release branch can be thought of being permanently "frozen" in this regard. Stable may be a better term, but it's already overloaded with a previous release branch or channel.
+- "Unfreezing" a branch or package.
+  - Allow for breaking changes to be made to branch or package.
+- "Critical packages"
+  - Important packages which have many dimensions of build or runtime behavior.
+  - Generally these will be packages which are a part of `stdenv`, or bootstrapping a system (e.g. systemd).
+  - Initially these will be: `stdenv.cc`,`binutils`, and `systemd` for this RFC.
```

Comment:
(nit) I guess at least `glibc` should be listed here as well.

Reply:
>  - Generally these will be packages which are a part of `stdenv`, or bootstrapping a system (e.g. systemd).

Separate building/bootstrapping stdenv from that of a (NixOS) system? I'm of the opinion all stdenv deps, including stdenv itself should freeze at this point.

Reply:
It was implied that this would include everything `stdenv.cc`, but I could make it more explicit.

```
nix-repl> stdenv.cc.libc.name
"glibc-2.32-25"

nix-repl> stdenv.cc.bintools.name
"binutils-wrapper-2.35.1"
```